plugins/chat/app/services/chat/create_category_channel.rb

```diff
@@ -68,13 +64,16 @@ def category_channel_does_not_exist(category:, contract:, **)
     end
 
     def create_channel(category:, contract:, **)
-      category.create_chat_channel(
-        name: contract.name,
-        slug: contract.slug,
-        description: contract.description,
-        user_count: 1,
-        auto_join_users: contract.auto_join_users,
-      )
+      channel =
+        category.create_chat_channel(
+          name: contract.name,
+          slug: contract.slug,
+          description: contract.description,
+          user_count: 1,
+          auto_join_users: contract.auto_join_users,
+        )
+      fail!(channel.errors.full_messages.join("\n")) if channel.errors.present?
```

Comment:
@Flink not sure how you feel about this and `on_failed_step` in the controller. I changed `create_channel` and `create_membership` to a step instead of a model, since the channel has its own AR validations that we need to check for on create.

Maybe there is a nicer way to handle this, like a `fail_model_validations!` method in the service/context? Not sure...